src/lib.rs

```diff
@@ -32,6 +32,25 @@ pub type UtransportListener = Box<dyn Fn(Result<UMessage, UStatus>) + Send + Syn
 
 const UATTRIBUTE_VERSION: u8 = 1;
 
+pub enum ZenohMode {
+    Peer,
+    Client,
+    Router,
+}
+
+/// Configuration struct for `UPClientZenoh`
+pub struct UPClientZenohConfig {
+    /// The Zenoh's mode (router, peer or client)
+    pub mode: ZenohMode,
+    /// Which endpoints to listen on. E.g. tcp/localhost:7447.
```

Comment:
My big question is "how clueless should the average uE developer be allowed to be?" :slightly_smiling_face: 

Should we write another API so that it "just works" as expected for 99% of cases where someone is writing a typical uE?

By that I mean -- if someone is writing a "typical" uE using `up-client-zenoh-rust`, they most likely are fine with having a default Zenoh Config, aside from having their mode set to Peer, right?

All of these boxed in pink:
![image](https://github.com/eclipse-uprotocol/up-client-zenoh-rust/assets/11622119/aceaa2ca-d040-4171-8030-b06d9839d4dd)

For those, perhaps a simpler API would be a little nicer, such as:
```rust
impl UPClientZenoh {
    pub fn new_peer() {
        let mut zenoh_config = Config::default();
        zenoh_config
            .set_mode(WhatAmI::Peer);
        UPClientZenoh::new_with_zenoh_config(zenoh_config).await
    }
}
```

Then we'd have:
1. Most simple API (`new_peer()`), just for making a typical Peer, that would be used by 99% of uE developers
2. Slightly more advanced API with enough flexibility for 99% of the remaining 1% (`new_with_config()`)
3. Most advanced API that gives access to all configurability of Zenoh (`new_with_zenoh_config`) for the remaining 0.01% of use cases

What do you think?

Reply:
In fact, the default config is using `Peer` mode and that's why I don't configure it explicitly. 😃

Reply:
I bow down before your API design haha 😆